ocaml/typing/typedtree.mli

```diff
@@ -163,7 +163,7 @@ and 'k pattern_desc =
             Invariant: n > 0
          *)
   | Tpat_array :
-      mutable_flag * Jkind.sort * value general_pattern list -> value pattern_desc
```

Comment:
I think middle-end or back-end might want to know the `m0` for arrays, or at least it's more correct to think that they do. In that case, it's better to make it available the typedtree, instead of make middle-end know that `m0` is always legacy for arrays.

Reply:
Yeah, ok, fair enough